app/controllers/admin/admin_users_controller.rb

```diff
@@ -39,76 +39,115 @@ def update_user
         flash[:error] = ts('There was an error updating user %{name}', :name => params[:id])
         redirect_to(request.env["HTTP_REFERER"] || root_path)
       end
-    elsif params[:admin_action]
-      @user = User.find_by_login(params[:user_login])
+    else
       @admin_note = params[:admin_note]
-      if @admin_note.blank?
-        flash[:error] = ts("You must include notes in order to perform this action")
-        redirect_to(request.env["HTTP_REFERER"] || root_path)
+      @user = User.find_by_login(params[:user_login])
+      submitted_kin_user = User.find_by_login(params[:next_of_kin_name])
+ 
+      # there is a next of kin username, but no email
+      if params[:next_of_kin_name].present? && params[:next_of_kin_email].blank?
+        flash[:error] = ts('Fannish next of kin email is missing.')
+        redirect_to request.referer || root_path
+
+      # there is a next of kin email, but no username
+      elsif params[:next_of_kin_name].blank? && params[:next_of_kin_email].present?
+        flash[:error] = ts('Fannish next of kin user is missing.')
+        redirect_to request.referer || root_path
+
+      # there is a next of kin username, but it is not a valid user
+      elsif params[:next_of_kin_name].present? && submitted_kin_user.nil?
+        flash[:error] = ts('Fannish next of kin user is invalid.')
+        redirect_to request.referer || root_path
+
+      # there is an admin action selected, but no note entered  
+      elsif params[:admin_action].present? && @admin_note.blank?
+        flash[:error] = ts('You must include notes in order to perform this action.')
+        redirect_to request.referer || root_path
+
+      # there is no length entered for the suspension
+      elsif params[:admin_action] == 'suspend' && params[:suspend_days].blank?
+        flash[:error] = ts('Please enter the number of days for which the user should be suspended.')
+        redirect_to request.referer || root_path
+
+      # we made it through without any errors, so let's do some stuff
       else
-        if params[:admin_action] == 'warn'
-          @user.create_log_item( options = {:action => ArchiveConfig.ACTION_WARN, :note => @admin_note, :admin_id => current_admin.id})
-          flash[:notice] = ts("Warning was recorded")
-          redirect_to(request.env["HTTP_REFERER"] || root_path)
-        elsif params[:admin_action] == 'suspend'
-          if params[:suspend_days].blank?
-            flash[:error] = ts("Please enter the number of days for which the user should be suspended.")
-            redirect_to(request.env["HTTP_REFERER"] || root_path)
+        success_message = []
+
+        # find or create a fannish next of kin if the fields are filled in
+        if params[:next_of_kin_name].present? && params[:next_of_kin_email].present?
+          if @user.fannish_next_of_kin.nil?
+            @user.fannish_next_of_kin = FannishNextOfKin.new(user_id: params[:user_login],
+              :kin_id => submitted_kin_user.id,
```

Comment:
Align the elements of a hash literal if they span more than one line.<br>Use the new Ruby 1.9 hash syntax.